server/game/core/ability/PlayerOrCardAbility.js

```diff
@@ -103,12 +103,11 @@ class PlayerOrCardAbility {
     }
 
     buildTargetResolver(name, properties) {
-        if (properties.mode === TargetMode.Select) {
-            return new SelectTargetResolver(name, properties, this);
-        } else if (properties.mode === TargetMode.Ability) {
-            return new AbilityTargetResolver(name, properties, this);
+        switch (properties.mode) {
+            case TargetMode.Select: return new SelectTargetResolver(name, properties, this);
```

Comment:
I've never seen one-line case statements like this before - did the linter force it? If not I'd recommend splitting them into two lines just for style consistency, I find it kind of hard to read like this